src/lib/dataset/mongo.ts

```diff
@@ -89,8 +90,13 @@ export function inferTypeFromDataset(dataset: DataSet, maxRows: number = 50): Da
       if (prevType === null) {
         prevType = guessedType;
       } else if (prevType !== guessedType) {
-        prevType = null;
-        break;
+        const typeSet = new Set([prevType, guessedType]);
```

Comment:
In this specific case, I think an explicit test like this one is easier to read:

```typescript
if ((prevType === 'float' && guessedType === 'integer') || (prevType === 'integer' && guesstedType === 'float')) {
    prevType = 'float';
}
```